Sources/SignalProducer.swift

```diff
@@ -2220,7 +2220,21 @@ extension SignalProducer where Error == NoError {
 	///
 	/// - returns: A producer that sends events from `self` and then from
 	///            `replacement` when `self` completes.
-	public func then<U>(_ replacement: SignalProducer<U, NoError>) -> SignalProducer<U, NoError> {
+	public func then<U, Replacement: SignalProducerConvertible>(_ replacement: Replacement) -> SignalProducer<U, NoError> where Replacement.Value == U, Replacement.Error == NoError {
+		return _then(replacement)
+	}
+	
+	/// Wait for completion of `self`, *then* forward all events from
+	/// `replacement`.
+	///
+	/// - note: All values sent from `self` are ignored.
+	///
+	/// - parameters:
+	///   - replacement: A producer to start when `self` completes.
+	///
+	/// - returns: A producer that sends events from `self` and then from
+	///            `replacement` when `self` completes.
+	public func then<Replacement: SignalProducerConvertible>(_ replacement: Replacement) -> SignalProducer<Value, NoError> where Replacement.Value == Value, Replacement.Error == NoError {
```

Comment:
Why is there a new variant?

Reply:
Probably because the type became ambiguous than before.
In the test, 4 errors occur including [this line](https://github.com/ReactiveCocoa/ReactiveSwift/blob/master/Tests/ReactiveSwiftTests/SignalProducerSpec.swift#L2207).